console/src/app/modules/providers/provider-gitlab/provider-gitlab.component.ts

```diff
@@ -0,0 +1,267 @@
+import { COMMA, ENTER, SPACE } from '@angular/cdk/keycodes';
+import { Location } from '@angular/common';
+import { Component, Injector, Type } from '@angular/core';
+import { AbstractControl, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatLegacyChipInputEvent as MatChipInputEvent } from '@angular/material/legacy-chips';
+import { ActivatedRoute } from '@angular/router';
+import { take } from 'rxjs';
+import {
+  AddGitLabProviderRequest as AdminAddGitLabProviderRequest,
+  GetProviderByIDRequest as AdminGetProviderByIDRequest,
+  UpdateGitLabProviderRequest as AdminUpdateGitLabProviderRequest,
+} from 'src/app/proto/generated/zitadel/admin_pb';
+import { Options, Provider } from 'src/app/proto/generated/zitadel/idp_pb';
+import {
+  AddGitLabProviderRequest as MgmtAddGitLabProviderRequest,
+  GetProviderByIDRequest as MgmtGetProviderByIDRequest,
+  UpdateGitLabProviderRequest as MgmtUpdateGitLabProviderRequest,
+} from 'src/app/proto/generated/zitadel/management_pb';
+import { AdminService } from 'src/app/services/admin.service';
+import { Breadcrumb, BreadcrumbService, BreadcrumbType } from 'src/app/services/breadcrumb.service';
+import { ManagementService } from 'src/app/services/mgmt.service';
+import { ToastService } from 'src/app/services/toast.service';
+
+import { PolicyComponentServiceType } from '../../policies/policy-component-types.enum';
+
+@Component({
+  selector: 'cnsl-provider-gitlab',
+  templateUrl: './provider-gitlab.component.html',
+  styleUrls: ['./provider-gitlab.component.scss'],
+})
+export class ProviderGitlabComponent {
+  public showOptional: boolean = false;
+  public options: Options = new Options();
+  public id: string | null = '';
+  public serviceType: PolicyComponentServiceType = PolicyComponentServiceType.MGMT;
+  private service!: ManagementService | AdminService;
+
+  public readonly separatorKeysCodes: number[] = [ENTER, COMMA, SPACE];
+
+  public form!: FormGroup;
+
+  public loading: boolean = false;
+
+  public provider?: Provider.AsObject;
+  public updateClientSecret: boolean = false;
+
+  constructor(
+    private route: ActivatedRoute,
+    private toast: ToastService,
+    private injector: Injector,
+    private _location: Location,
+    private breadcrumbService: BreadcrumbService,
+  ) {
+    this.form = new FormGroup({
+      name: new FormControl('', []),
+      clientId: new FormControl('', [Validators.required]),
+      clientSecret: new FormControl('', [Validators.required]),
```

Comment:
here too

Reply:
I'd prefer to merge this before #5409 as these validators can be changed there too and there are still some things to be fixed 👍